p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -141,23 +145,24 @@ where
         self.addresses.keys()
     }
 
-    /// Selects peer addresses for outbound connections (except reserved)
-    pub fn select_new_outbound_addresses(
-        &self,
-        pending_outbound: &BTreeSet<A>,
-        connected_outbound_count: usize,
-    ) -> Vec<A> {
+    /// Selects peer addresses for outbound connections (except reserved).
+    /// Only one outbound connection is allowed per address group.
+    pub fn select_new_outbound_addresses(&self, all_outbound: &BTreeSet<A>) -> Vec<A> {
         let now = self.time_getter.get_time();
-        let count = MAX_OUTBOUND_CONNECTIONS
-            .saturating_sub(pending_outbound.len())
-            .saturating_sub(connected_outbound_count);
+        let count = MAX_OUTBOUND_CONNECTIONS.saturating_sub(all_outbound.len());
```

Comment:
Shouldn't we just return if this is zero?

Reply:
Good idea, thanks.